File: dannybrown/src/day1.py
from helpers import read_puzzle_input

def day1_part1(depths):
    length = len(depths)
    sum = 0
    for i in range(length):
      if i == length - 1:
        break
      sum += depths[i] < depths[i+1]

    return sum

def day1_part2(depths):
  length = len(depths)
  sum = 0

  for i in range(len(depths)):
    if i + 2  == length - 1:
      break

    first_batch = depths[i] + depths[i+1] + depths[i+2]
    second_batch = depths[i+1] + depths[i+2] + depths[i+3]

    if first_batch < second_batch:
      sum += 1
  return sum

if __name__ == '__main__':
  depths = read_puzzle_input("dannybrown/input/day1.txt")

  print("Day 1 Part 1:", day1_part1(depths))
  print("Day 2 Part 2:", day1_part2(depths))

File: dannybrown/src/day1/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)
func check(e error) {
	if e != nil {
		panic(e)
	}
}

func day1_part1(input []string) int {
	var sum int
	var inputLength int = len(input)

	for i := 0; i < inputLength; i++ {
		if i == inputLength - 1 {
			break
		}
		val1, _ := strconv.Atoi(input[i])
		val2, _ := strconv.Atoi(input[i+1])
		if val1 < val2 {
			sum++
		}
	}
	return sum
}

func day1_part2(input []string) int {
	var sum int
	var inputLength int = len(input)

	for i := 0; i < inputLength; i++ {
		if i + 2 == inputLength - 1 {
			break
		}
		val1, _ := strconv.Atoi(input[i])
		val2, _ := strconv.Atoi(input[i+1])
		val3, _ := strconv.Atoi(input[i+2])
		val4, _ := strconv.Atoi(input[i+3])

		firstBatch := val1 + val2 + val3
		secondBatch := val2 + val3 + val4
		if firstBatch < secondBatch {
			sum++
		}
	}
	return sum
}

func main() {
	dat, err := os.Open("../input/day1.txt")
	check(err)
	defer dat.Close()
	scanner := bufio.NewScanner(dat)
	scanner.Split(bufio.ScanLines)
	var txtlines []string

	for scanner.Scan() {
		txtlines = append(txtlines, scanner.Text())
	}

	fmt.Println(day1_part1(txtlines)) // Part 1
	fmt.Println(day1_part2(txtlines)) // Part 2
}
